Added AMS to the stack

batatasfritas opened the pull request:
Included group_restricted.lua since we limite access
Included load intel intel-oneapi-mpi since it depends on the stack's IMPI

batatasfritas committed:
Added AMS to the stack

Included group_restricted.lua since we limite access
Included load intel intel-oneapi-mpi since it depends on the stack's IMPI

batatasfritas commented:
I'm not sure whether the load functionality works.

I took this from the spack-packagelist, since it's more or less essential in this case.

nrichart commented:
Why not mark is as despendent of intel intel-oneapi-mpi in the spec, and then it will appear in the module hierarchy
Other why it will appear in core but in fact depend on intel

nrichart commented on stacks/syrah/syrah.yaml:
```diff
@@ -615,6 +615,11 @@ external_packages:
         externals:
           - spec: adf@2017.11 %<core_compiler>
             prefix: <external_prefix>/adf
+    - ams:
```
It has to be in a list that depends on `intel` `intel-oneapi-mpi`

batatasfritas replied:
So, we move it to `mpi_packages_intel_stable`?

nrichart commented on stacks/syrah/syrah.yaml:
```diff
@@ -615,6 +615,11 @@ external_packages:
+        default: {buildable: false }
+        externals:
+          - spec: ams@2022.103 %<core_compiler>
+            prefix: <external_prefix>/ams
```
if you use `group_permission` you also need an entry of the type
```
permissions:
  read: group
  group: comsol-soft
```

batatasfritas commented:
This is something I was just discussing with Eduardo.

We can opt for that, yes. I did it this way to minimize the impact to the users, since currently `adf` is in core (and it was, itself, also dependent on intel-mpi.
If we decide to put it as dependent of `intel` `intel-oneapi-mpi` I'm fine with it.

batatasfritas committed:
Moved ams to mpi_packages_intel_stable, added the permissions section

batatasfritas commented:
OK, I did the changes. I'm unsure how a `externals` will fit in one of the `mpi_packages_intel_stable` so feedback highly appreciated.

nrichart commented:
So `adf` is not working in fact should it be removed or at least corrected ?

nrichart commented on stacks/syrah/syrah.yaml:
```diff
@@ -777,6 +777,15 @@ mpi_packages_intel_stable:
+            read: group
+            group: adf-soft
+        externals:
+          - spec: ams@2022.103 %<core_compiler>
```
You should remove `%<core_compiler>` and either put nothing or specify `%intel`

batatasfritas replied:
Thanks for spotting that.

nrichart commented:
The location in the file defines the `spec` that will be installed in this case we want `ams %intel ^intel-oneapi-mpi` and the externals add some description to the `packeges.yaml` saying to install this spec do nothing but it is installed already in the specified `prefix`

batatasfritas commented:
ADF worked on Helvetios. With the update to `syrah` it did stop working (unless the user adds `intel-oneapi-mpi` to their list of modules).
Since there was a major change in the input structure in between these two versions, I suggest we correct ADF and let it be in the stack for a little longer (so users can finish whatever projects). On the next stack we remove it.

batatasfritas committed:
Corrected the core compiler issue, repeated the same structure for ADF (since all the same story applies)

batatasfritas commented:
I added ADF to this, since all the same applies and we avoid wasting time doing it a second time. Whatever is wrong in one, is wrong in the other now and I'll fix both at the same time.

nrichart reviewed (approved):
LGTM

batatasfritas committed:
Fixed prefixes for ADF and AMS.

Currently ADF_ROOT points to <external>/adf instead of <external>/adf/adf2017.111.

batatasfritas commented:
Added a correction since the prefix in `syrah.yaml` produces a module with the wrong `{ADF,AMS}_ROOT`. This is rarely used in practice, doing the change for consistency with the rest of our stack.
